distil/primitives/list_encoder.py

```diff
@@ -0,0 +1,143 @@
+import os
+from typing import List, Set, Any, Sequence
+import logging
+
+from d3m import container, utils
+from d3m.metadata import base as metadata_base, hyperparams, params
+from d3m.primitive_interfaces import base, unsupervised_learning
+
+import pandas as pd
+
+logger = logging.getLogger(__name__)
+
+__all__ = ('ListEncoderPrimitive',)
+
+
+class Hyperparams(hyperparams.Hyperparams):
+    use_columns = hyperparams.Set(
+        elements=hyperparams.Hyperparameter[int](-1),
+        default=(),
+        semantic_types=['https://metadata.datadrivendiscovery.org/types/ControlParameter'],
+        description="A set of column indices to force primitive to operate on. If any specified column cannot be parsed, it is skipped.",
+    )
+
+
+class Params(params.Params):
+    pass
+
+
+class ListEncoderPrimitive(unsupervised_learning.UnsupervisedLearnerPrimitiveBase[
+                               container.DataFrame, container.DataFrame, Params, Hyperparams]):
+    """
+     List Encoder takes columns that are made up out of lists and replaces them by expanding the list
+    across multiple columns.
+    """
+
+    metadata = metadata_base.PrimitiveMetadata(
+        {
+            'id': 'd3d421cb-9601-43f0-83d9-91a9c4199a06',
+            'version': '0.1.0',
+            'name': "List encoder",
+            'python_path': 'd3m.primitives.data_transformation.list_encoder.DistilListEncoder',
+            'source': {
+                'name': 'Distil',
+                'contact': 'mailto:cbethune@uncharted.software',
+                'uris': [
+                    'https://github.com/uncharted-distil/distil-primitives/distil/primitives/list_encoder.py',
+                    'https://github.com/uncharted-distil/distil-primitives',
+                ],
+            },
+            'installation': [{
+                'type': metadata_base.PrimitiveInstallationType.PIP,
+                'package_uri': 'git+https://github.com/uncharted-distil/distil-primitives.git@{git_commit}#egg=distil-primitives'.format(
+                    git_commit=utils.current_git_commit(os.path.dirname(__file__)),
+                ),
+            }],
+            'algorithm_types': [
+                metadata_base.PrimitiveAlgorithmType.DATA_CONVERSION,
```

Comment:
Note that we can make PRs against `definitions.json` if we want to add missing types.  `DATA_CONVERSION` is fine here, but if you wanted to be more explicit, you could add a LIST_ENCODER type..